core/src/main/resources/jenkins/model/Jenkins/configure.jelly

```diff
@@ -40,11 +40,11 @@ THE SOFTWARE.
         ${it.rootDir}
       </f:entry>
       <f:advanced>
-        <f:entry field="rawWorkspaceDir" title="${%Workspace Root Directory}">
-          <f:textbox/>
+        <f:entry field="rawWorkspaceDir" title="${%Workspace Root Directory}" >
+          <f:textbox readonly="readonly" />
         </f:entry>
         <f:entry field="rawBuildsDir" title="${%Build Record Root Directory}">
-          <f:textbox/>
+          <f:textbox readonly="readonly"/>
```

Comment:
Setting the `readonly` attribute will block a web browser from editing this field, but `doConfigSubmit` is continuing to accept modifications from HTTP posts! I think you want to delete those two lines as well.

I would really just suggest deleting this `advanced` block, and print a warning to the system log if there is an override which is not matched by the system property.

Reply:
done it all